platform/eventing/listener.go

```diff
@@ -0,0 +1,97 @@
+package eventing
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/cloudevents/sdk-go/v2/client"
+	"github.com/cloudevents/sdk-go/v2/event"
+	cloudeventsnats "github.com/cloudevents/sdk-go/v2/protocol/nats"
+	"github.com/fewlinesco/go-pkg/platform/monitoring"
+	"github.com/jmoiron/sqlx"
+)
+
+var (
+	ErrConsumerEventCanNotBePersisted = errors.New("the event could not be persisted in the database")
+)
+
+type listenerSubject string
+
+// Listener subscribes to subjects/queues in the broker and persists them in a local database
+type Listener struct {
+	URL      string
+	subjects []string
+	db       *sqlx.DB
+	logger   *log.Logger
+
+	maxNumberOfRetries int
+}
+
+// NewListener creates a new config for a listener
+func NewListener(URL string, subjects []string, db *sqlx.DB, logger *log.Logger) *Listener {
+	return &Listener{
+		URL:      URL,
+		subjects: subjects,
+		db:       db,
+		logger:   logger,
+
+		maxNumberOfRetries: 5,
+	}
+}
+
+// Start starts a listener for the provided subjects
+// It will save all events for the subjects in the DB
+func (listener *Listener) Start() {
+	for _, subject := range listener.subjects {
+		ctx := context.WithValue(context.Background(), listenerSubject(subject), subject)
+
+		natsConsumer, err := cloudeventsnats.NewConsumer(listener.URL, subject, cloudeventsnats.NatsOptions())
+		if err != nil {
+			listener.logger.Printf("failed to create nats consumer, %v", err)
+			os.Exit(1)
```

Comment:
I think this should return an error instead of calling `os.Exit` and a logger. As a rule I think only `main` should be able to call `os.Exit`. Moreover it means we can skip the log part as `main` is in charge of handling or log the error